prow/kube/config_test.go

```diff
@@ -0,0 +1,673 @@
+			if err != nil {
+				t.Fatalf("%s: could not create kubeconfig file: %v", testCase.name, err)
+			}
+			defer func() {
+				if err := os.Remove(kubeconfig.Name()); err != nil {
+					t.Fatalf("%s: failed to clean up temp file: %v", testCase.name, err)
+				}
+			}()
+			if _, err := kubeconfig.WriteString(testCase.kubeconfig); err != nil {
+				t.Fatalf("%s: could not populate kubeconfig file: %v", testCase.name, err)
+			}
+
+			// we either read this explicitly through the flag or implictly through the env
+			var name string
+			if testCase.explicitPath {
+				name = kubeconfig.Name()
+			} else {
+				if err := os.Setenv("KUBECONFIG", kubeconfig.Name()); err != nil {
+					t.Fatalf("%s: could not populate $KUBECONFIG env: %v", testCase.name, err)
+				}
+			}
+			configurations, defaultContext, err := kubeconfigConfigLoader(name)()
+			if !testCase.expectedErr && err != nil {
+				t.Errorf("%s: expected no error but got one: %v", testCase.name, err)
+			}
+			if testCase.expectedErr && err == nil {
+				t.Errorf("%s: expected an error but got none", testCase.name)
+			}
+			if !reflect.DeepEqual(configurations, testCase.expected) {
+				t.Errorf("%s: got incorrect cluster configurations: %s", testCase.name, diff.ObjectReflectDiff(configurations, testCase.expected))
+			}
+			if !reflect.DeepEqual(defaultContext, testCase.expectedDefault) {
+				t.Errorf("%s: got incorrect default context: %s", testCase.name, diff.ObjectReflectDiff(defaultContext, testCase.expectedDefault))
+			}
+		})
+	}
+}
+
+func TestBuildClusterConfigLoader(t *testing.T) {
+	var testCases = []struct {
+		name         string
+		buildCluster string
+
+		expected        map[string]rest.Config
+		expectedDefault *string
+		expectedErr     bool
+	}{
+		{
+			name: "implicit build cluster loads fine and sets no default",
+			buildCluster: `endpoint: https://some.cluster.com:443
+clientCertificate: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNIakNDQWNnQ0FRRXdEUVlKS29aSWh2Y05BUUVMQlFBd2daa3hDekFKQmdOVkJBWVRBbFZUTVJNd0VRWUQKVlFRSURBcFhZWE5vYVc1bmRHOXVNUkF3RGdZRFZRUUhEQWRUWldGMGRHeGxNUkF3RGdZRFZRUUtEQWRTWldRZwpTR0YwTVJJd0VBWURWUVFMREFsUGNHVnVVMmhwWm5ReEdUQVhCZ05WQkFNTUVITnZiV1V1WTJ4MWMzUmxjaTVqCmIyMHhJakFnQmdrcWhraUc5dzBCQ1FFV0UzTnJkWHB1WlhSelFISmxaR2hoZEM1amIyMHdIaGNOTVRrd01qRXcKTWpNME9UQXlXaGNOTWpBd01qRXdNak0wT1RBeVdqQ0JtVEVMTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnTQpDbGRoYzJocGJtZDBiMjR4RURBT0JnTlZCQWNNQjFObFlYUjBiR1V4RURBT0JnTlZCQW9NQjFKbFpDQklZWFF4CkVqQVFCZ05WQkFzTUNVOXdaVzVUYUdsbWRERVpNQmNHQTFVRUF3d1FjMjl0WlM1amJIVnpkR1Z5TG1OdmJURWkKTUNBR0NTcUdTSWIzRFFFSkFSWVRjMnQxZW01bGRITkFjbVZrYUdGMExtTnZiVEJjTUEwR0NTcUdTSWIzRFFFQgpBUVVBQTBzQU1FZ0NRUUMyOThDSXBJVzFDUUZ1clczWVNjTVNMSGx5V1JZNXozY0JuSXRFT1ErMWZuLzU3NmZtCk5Ha3pXemxKcXVPWFNVMlNtdytrUFlha3l6ZHFCRHZBRzBiakFnTUJBQUV3RFFZSktvWklodmNOQVFFTEJRQUQKUVFDUUZMYVkvRUpNRkVCQllGRkIrUUhSOXdMcVlaSW0ydEpKd0grSXNEVGIxd0xnWE5JSHBpbjZ1SzIxdmJwbgo5YXU4alBUUWRXSGFrQTdKSUNLSytxYzUKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
```

Comment:
Real values are awfully long. Instead can we do something more useful for testing like: `clientCertificate: aHR0cHM6Ly9zb21lLmNsdXN0ZXIuY29tOjQ0MyBjbGllbnQgY2VydAo=`

```console
$ echo "https://some.cluster.com:443 client cert" | base64   
aHR0cHM6Ly9zb21lLmNsdXN0ZXIuY29tOjQ0MyBjbGllbnQgY2VydAo=
```

Reply:
I think I tried and something failed since it was not actual data, let me try agin